server/app/serializers/rpc/service_pod_serializer.rb

```diff
@@ -72,6 +72,16 @@ def build_secrets
         secrets << {name: "SSL_CERTS", type: 'env', value: domain_auth.tls_sni_certificate}
       end
 
+      # Inject certificates as secrets
+      service.certificates.each do |certificate|
+        grid_cert = grid.certificates.find_by(subject: certificate.subject)
+        item = {name: certificate.name, type: certificate.type, value: nil}
+        if grid_cert
+          item[:value] = [grid_cert.certificate, grid_cert.private_key].join
```

Comment:
The current `LE_CERTIFICATE_example.com_PRIVATE_KEY/CERTIFICATE/BUNDLE` allows using the key and cert separately, and I imagine some applications expect to be configured with the key/cert as separate files/params, and possibly even the chain as a third parameter (like [Apache < 2.4.8](https://httpd.apache.org/docs/current/mod/mod_ssl.html#sslcertificatechainfile)). Is this kind of cert+key bundle specific to haproxy?

I'm thinking this should have some kind of parametrization to allow deploying the cert/key/chain separately, in order to support a variety of applications, not just `kontena/lb`. I don't know what that YAML would look like... 

```yaml
certificates:
  - subject: example.com
    cert_env: SSL_CERT
    key_env: SSL_KEY
    chain_env: SSL_CHAIN
```

Reply:
> Is this kind of cert+key bundle specific to haproxy?

Yes, of course there might be other systems using the same mechanism too but HAProxy has worked like this as long as I remember using it

> I'm thinking this should have some kind of parametrization to allow deploying the cert/key/chain separately

Not a bad idea at all. It just needs option in yaml for all four:
- private key
- cert
- chain
- bundle (key + cert)


IMO the yaml you posted looks pretty ok, maybe not good to have the option called `xyz_env` as we might have someday support for injecting these through files also. Maybe `cert_var` would be more future proof? 

The typical use case when using Kontena LB would then be:
```
certificates:
  - subject: example.com
    bundle_var: SSL_CERT
    type: env
```

@jakolehm Any thoughts if we'd go to this direction?

Reply:
Important point is that for the separate cert+key case, you need both from the same cert, and I reckon you should be able to have both of those in the same item.

```yaml
certificates:
  - subject: example.com
    type: env
    key_name: SSL_KEY
    cert_name: SSL_CERT
```

Reply:
I think current implementation (bundle) is ok default.. we can improve this later in a separate PR.